be/src/exec/parquet_reader.cpp

```diff
@@ -166,6 +176,94 @@ Status ParquetReaderWrap::init_parquet_reader(const std::vector<SlotDescriptor*>
     }
 }
 
+Status ParquetReaderWrap::get_schema(std::vector<SlotDescriptor>* schema) {
+    RETURN_IF_ERROR(_init_parquet_reader());
+
+    const auto& file_schema = _file_metadata->schema();
+
+    for (auto i = 0; i < file_schema->group_node()->field_count(); i++) {
+        const auto &field = file_schema->group_node()->field(i);
+        const auto &name = field->name();
+
+        if (!field->is_primitive()) {
+            // Now, we treat all nested types as VARCHAR.
+            schema->emplace_back(i, name, TypeDescriptor::create_varchar_type(TypeDescriptor::MAX_VARCHAR_LENGTH));
+            continue;
+        }
+
+        auto column = file_schema->Column(file_schema->ColumnIndex(*field));
+
+        auto physical_type = column->physical_type();
+        auto logical_type = column->logical_type();
+
+        // See detail in https://arrow.apache.org/docs/cpp/parquet.html
+        TypeDescriptor tp;
+        switch (physical_type) {
+        case parquet::Type::BOOLEAN:
+            tp = TypeDescriptor(TYPE_BOOLEAN);
+            break;
+        case parquet::Type::FLOAT:
+            tp = TypeDescriptor(TYPE_FLOAT);
+            break;
+        case parquet::Type::DOUBLE:
+            tp = TypeDescriptor(TYPE_DOUBLE);
+            break;
+        case parquet::Type::INT32:
+            if (logical_type->is_int()) {
+                tp = TypeDescriptor(TYPE_INT);
+            } else if (logical_type->is_date()) {
+                tp = TypeDescriptor(TYPE_DATE);
+            } else if (logical_type->is_time()) {
+                tp = TypeDescriptor(TYPE_TIME);
+            } else if (logical_type->is_decimal()) {
+                tp = TypeDescriptor(TYPE_DECIMAL);
+            } else {
+                tp = TypeDescriptor(TYPE_INT);
```

Comment:
Seems the Logical type of `int` have been handled at line 212-213.

Returns an error status here?

Reply:
The column with physical type INT32 probably has no logical type. The default branch is necessary here.